CMakeLists.txt

```diff
@@ -86,8 +86,6 @@ string(REPLACE ":" ";" TEST_CLASSPATH_LIST "${TEST_CLASSPATH}")
 
 # Needed as we abuse some of the test compile macros to test shared lib links
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-# CMake's default RPath handling includes the full build directory path, which isn't needed. Turn it off.
-set(CMAKE_SKIP_RPATH TRUE)
```

Comment:
If I remember correctly, before I added this line, the RPath of `libamazonCorrettoCryptoProvider.so` on my Ubuntu desktop contained both of the following when running `chrpath libamazonCorrettoCryptoProvider.so`:
1. `/home/aweibel/workspace/github/amazon-corretto-crypto-provider/build/awslc/bin/lib/libcrypto.so` (The path I where I was building ACCP on my desktop.)
2. `$ORIGIN`

Adding these 2 lines caused the 1st RPath value of the temporary build directory to be removed from the resulting  output, since only the 2nd RPath value is needed. Deleting these two lines **might** re-add the temporary build path's back into the libACCP on Linux platforms, and this should be checked before merging. Theoretically, if there's an extra RPath value containing a hard-coded directory from where our CI is configured to build and publish ACCP from, then any user on a host that is using ACCP could recreate that hard-coded path and put a different `libcrypto.so` file there that would loaded into any Java processes using ACCP on that host.

We may want to add a check during the build that ensures that only 1 RPath value is present in `libamazonCorrettoCryptoProvider.so`, and that it's value is `$ORIGIN`.

Reply:
nice catch. looks like deleting this line does indeed introduce multiple rpath entries (albeit in the order we want)

```
root@817bfef6156a:/accp# cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
root@817bfef6156a:/accp# chrpath -l ./build/cmake/libamazonCorrettoCryptoProvider.so
./build/cmake/libamazonCorrettoCryptoProvider.so: RUNPATH=$ORIGIN:/accp/build/awslc/bin/lib
```

agreed that rpath should have one entry and point to `$ORIGIN`. i'll walk this back and start looking at ways to prune the rpath programmatically on MacOS and also look into a build-time check that asserts our rpath requirements.

Reply:
ok, digging into it further, it looks like the root cause for lack of MacOS support before this commit was that rpath value of `$ORIGIN` doesn't mean anything on MacOS. As shown above, my prior approach included the aws-lc build dir in rpath, which i think is where `libcrypto.so` was being found in the initial draft of this PR. the fix was to use the MacOS equivalent of `$ORIGIN`, which is [`@loader_path`](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/DynamicLibraryDesignGuidelines.html#//apple_ref/doc/uid/TP40002013-SW21). I've confirmed that in this PR's current version, there's only one RPATH entry (see testing section in the PR overview) and its set appropriately for mac and linux.

i also looked into adding a build-time check for this fact, but my CMake-fu is rather weak when it comes to invoking compound commands. in bash with `-e` and `-pipefail` set, i think the checks would look something like this:

mac:

```
# check that there's only one rpath value (probably not the best way to say "expect only one matching line")
[[ $(echo 1) == $(otool -l ./path/to/libamazonCorrettoCryptoProvider.dylib | ' cmd LC_RPATH' | wc -l)

# check that it's set to @loader_path
otool -l ./path/to/libamazonCorrettoCryptoProvider.dylib | grep -A2 ' cmd LC_RPATH' | tail -1 | grep 'libamazonCorrettoCryptoProvider'
```

linux:

```
# check that there's only one rpath value and that it's set to @loader_path
chrpath -l ./path/to/libamazonCorrettoCryptoProvider.so | grep -P 'RUNPATH=\$ORIGIN$'
```

Reply:
test check added in most recent revision.